### PR TITLE
Vigenere Cipher in Java

### PR DESCRIPTION
**Pull request by parjanyaacoder**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Vigenere Cipher algo added in Java.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Solves #974

### CODE REVIEW

**Comment by parjanyaacoder** [comment]
@iamrajiv @siddharth25pandey @HarshCasper  Please review.

File: Java/cryptography/vigenereCipher.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+
```
Comment (HarshCasper) [inline Java/cryptography/vigenereCipher.java:1]: Add a proper description of the Code along with Examples for I/O

File: Java/cryptography/vigenereCipher.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        ch=(char)(ch-26);
+        s1 = s1 + ch;
+        indexJ++;
+    }
+      return s1;
+}
+
+public static String decrypt(String str,String key)
+{
+    int indexI= 0,indexJ=0;String s1="";char ch;
```
Comment (HarshCasper) [inline Java/cryptography/vigenereCipher.java:23]: Use Meaningful Variable Names

File: Java/cryptography/vigenereCipher.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    for(indexI=0;indexI<str.length();indexI++)
+    {
+        if(indexJ>=key.length())
+        indexJ=0;
+        ch = (char)((str.charAt(indexI)-(key.charAt(indexJ)-65)%26));
+        if(ch<'A')
+        ch=(char)(ch+26);
```
Comment (HarshCasper) [inline Java/cryptography/vigenereCipher.java:30]: Format your Code properly

File: Java/cryptography/vigenereCipher.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    return s1;
+}
+
+    
+public static void main(String[] args)
+{
+    Scanner sc = new Scanner(System.in);
+    String str = sc.nextLine();
+    String key = sc.nextLine();
+   String encrpytedString = encrypt(str.toUpperCase(),key.toUpperCase());
```
Comment (HarshCasper) [inline Java/cryptography/vigenereCipher.java:43]: Do proper Indentation

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make these changes

**Pull request closed without merging**